### DISCUSSION TITLE
Enabeling `globalReturn` in ESLint 9 with flat config.

### DISCUSSION THREAD

**Discussion by robertsosinski** [eslint/eslint#19197] [Q&A]
Hello y'all,

I'm using ESLint 9.14 in [Astro](https://astro.build) and I need to return within the global scope within `.astro` files, which looks like `return Astro.redirect('/go/here')`. However, ESLint sees this as using an global return and raises the following error `Parsing error: 'return' outside of functioneslint`.

So, my file looks something like this.
```astro
---
if (Astro.params.error) {
  return Astro.redirect('/go/here');
}
---
<h1>Content Here</h1>
```

And the return is totally fine in this file, but ESLint sees it as an error. My ESLint config file looks like this. Any help on how I can enable global returns would really help.

```javascript
import globals from 'globals';
import pluginJs from '@eslint/js';
import stylistic from '@stylistic/eslint-plugin';
import pluginAstro from 'eslint-plugin-astro';

export default [
  {
    ignores: ['dist', 'public'],
  },
  {
    languageOptions: {
      globals: {
        ...globals.browser,
        ...globals.node,
      },
      parserOptions: {
        ecmaFeatures: {
          globalReturn: true,
        },
      },
    },
  },
  pluginJs.configs.recommended,
  stylistic.configs.customize({
    semi: true,
    jsx: false,
  }),
  ...pluginAstro.configs.recommended,
];
```

Thanks!

**Comment by nzakas** [comment]
I'm guessing that `eslint-plugin-astro` is using a custom parser, so you should look at the documentation for that plugin to find the answer. This is a parser-level setting, not an ESLint-level setting.
